libs/SalesforceSDKCore/SalesforceSDKCore/Classes/RestAPI/SfapApi/SfapApiClient.swift

```diff
@@ -0,0 +1,149 @@
+/*
+ SfapApiClient.swift
+ SalesforceSDKCore
+ 
+ Created by Eric C. Johnson (Johnson.Eric@Salesforce.com) on 20250108.
+ 
+ Copyright (c) 2025-present, salesforce.com, inc. All rights reserved.
+ 
+ Redistribution and use of this software in source and binary forms, with or without modification,
+ are permitted provided that the following conditions are met:
+ * Redistributions of source code must retain the above copyright notice, this list of conditions
+ and the following disclaimer.
+ * Redistributions in binary form must reproduce the above copyright notice, this list of
+ conditions and the following disclaimer in the documentation and/or other materials provided
+ with the distribution.
+ * Neither the name of salesforce.com, inc. nor the names of its contributors may be used to
+ endorse or promote products derived from this software without specific prior written
+ permission of salesforce.com, inc.
+ 
+ THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND ANY EXPRESS OR
+ IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND
+ FITNESS FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT OWNER OR
+ CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL
+ DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
+ DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY,
+ WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY
+ WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+import Foundation
+
+@objc(SFApApiClient)
+public class SfapApiClient : NSObject {
+    
+    /// The error domain
+    static let errorDomain = "com.salesforce.sfap_api.client"
+    
+    /// The sfap_api hostname
+    private let apiHostName: String
+    
+    /// The sfap_api model name
+    private let modelName: String?
+    
+    /// The REST client
+    private let restClient: RestClient
+    
+    @objc
+    public init(
+        apiHostName: String,
+        modelName: String? = nil,
+        restClient: RestClient) {
+            self.apiHostName = apiHostName
+            self.modelName = modelName
+            self.restClient = restClient
+        }
+    
+    /// Fetches generated text from the `sfap_api` "generations" endpoint.
+    /// - Parameters:
+    ///   -  prompt: The prompt request parameter value
+    /// - Returns: The endpoint's response
+    @objc
+    public func fetchGeneratedText(
+        _ prompt: String
+    ) async throws -> SfapApiGenerationsResponseBody {
+        
+        // Guards.
+        guard
+            let userAccountCurrent = UserAccountManager.shared.currentUserAccount,
+            let restClient = RestClient.restClient(for: userAccountCurrent) else {
```

Comment:
Why is this getting a RestClient based on the current user vs using the RestClient from init?